config.go

```diff
@@ -686,6 +692,26 @@ func ValidateConfig(cfg Config, usageMessage string,
 		return nil
 	}
 
+	// IsSet returns true if an option has been set in either the config
+	// file or by a flag.
```

Comment:
Hmm, shouldn't this already be handled by our current behavior, where we read the command line options, then attempt to merge/override them with what we read from the config itself?

Reply:
It should, but right now we can not distinguish between set and not set boolean parameters. There is no way to find out if 
 `sync-freelist` has been explicitly set, except through the go-flag parsing object. Therefore we don't know if we should override the value of `db.bolt.nofreelistsync` or not. This is the reason the issue occurs (as stated in #5519).

Reply:
I tried my hardest to find a way around it without invoking the parsing objects `IsSet` function but I couldn't.

Reply:
I think this is not needed?
If `SyncFreeList` is not set then value is `false` so we get the same as the `Bolt.NoFreeListSync` default so overriding has no effect.
If `SyncFreeList` is set then the value is true we want to override.

Reply:
@bhandras Problem is right here:
https://github.com/lightningnetwork/lnd/blob/4487acc5d9ae3c0074bfc407825b15c6e0da1c24/config.go#L1443
If we have (in lnd.conf)...
```
[Application Options]
sync-freelist=false

[bolt]
db.bolt.nofreelistsync=true
```
...it will ignore `sync-freelist=false` since `sync-freelist` is superordinate `db.bolt.nofreelistsync`. I agree, I would rather do this without the new function `IsSet`, but could not find a way around it with go-flags.

Reply:
Indeed, missed the precedence aspect. Thanks for the thorough explanation @ErikEk